File: liveFPScount.cpp
#include <iostream>
#include "opencv2/opencv.hpp"
#include <time.h>

using namespace cv;
using namespace std;

int main(int argc, char** argv)
{
    Ptr<BackgroundSubtractor> pBackSub;
    pBackSub = createBackgroundSubtractorKNN();

    // Start default camera
    VideoCapture video(0);

    double fps = video.get(CAP_PROP_FPS);
    cout << "Frames per second camera : " << fps << endl;

    // Number of frames to capture
    int num_frames = 1;

    // Start and end times
    clock_t start;
    clock_t end;

    Mat frame, fgMask;

    cout << "Capturing " << num_frames << " frames" << endl;

    double ms, fpsLive;

    //chrono::steady_clock::time_point start, end;
    //std::common_type_t<std::chrono::steady_clock::duration, std::chrono::steady_clock::duration> diff;

    int keyboard;

    while(true){


        // Start time
        start = clock();
        //start = chrono::steady_clock::now();

        video >> frame;

        pBackSub->apply(frame, fgMask);

        erode(fgMask, fgMask, (5, 5));
        dilate(fgMask, fgMask, (5, 5));

        long sum = 0;
        int N = 1;

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                sum += 1;
            }
        }

        // End Time
        end = clock();
        //end = chrono::steady_clock::now();

        //diff = end - start;

        //cout << "start: " << start << endl;
        //cout << "end: " << end << endl;

        // Time elapsed
        double seconds =  (double(end) - double(start)) / double(CLOCKS_PER_SEC);
        cout << "Time taken : " << seconds << " seconds" << endl;
        //ms = chrono::duration <double, milli>(diff).count();
        //cout << ms << " ms" << endl;

        // Calculate frames per second
        fpsLive = double(num_frames) / double(seconds);
        //fpsLive = double(num_frames) / double(ms*0.001);
        cout << "Estimated frames per second : " << fpsLive << endl;

        putText(fgMask, "FPS: " + to_string(int(fpsLive)), { 50, 50 }, FONT_HERSHEY_SIMPLEX, 1.5, (255, 255, 255),2);

        imshow("fgMask", fgMask);

        keyboard = waitKey(30);
        if (keyboard == 'q' or keyboard == 27) {
            break;
        }
    }


    // Release video
    video.release();
    return 0;
}

File: liveFPScount.py
import cv2
import numpy as np
import time

# Background subtraction algortihm
backSub = cv2.createBackgroundSubtractorKNN()



# Start camera
video = cv2.VideoCapture(0);


fps = video.get(cv2.CAP_PROP_FPS)
print("Frames per second camera: {0}".format(fps))

# Number of frames to capture
num_frames = 1;

print("Capturing {0} frames".format(num_frames))


# Grab a few frames
while True:

    # Start time
    start = time.time()

    ret, frame = video.read()

    if frame is None:
        print("No frame")
        break

    fgMask = backSub.apply(frame)

    sum = 0
    N = 300
    for i in range(0, N):
        for j in range(0, N):
            sum += 1

    kernel = np.ones((5,5), np.uint8)

    fgMask = cv2.erode(fgMask, kernel, iterations=1)
    fgMask = cv2.dilate(fgMask, kernel, iterations=1)


    # End time for whole program running 120 frames
    end = time.time()

    # Time elapsed
    seconds = end - start
    #print ("Time taken : {0} seconds".format(seconds))

    # Calculate frames per second
    fps  = num_frames / seconds
    #print("Estimated frames per second : {0}".format(fps))

    cv2.putText(fgMask, "FPS: " + str(round(fps)), (50,50), cv2.FONT_HERSHEY_SIMPLEX, 1.5, (255,255,255))

    cv2.imshow('fgMask', fgMask)



    keyboard = cv2.waitKey(30)
    if keyboard == 'q' or keyboard == 27:
        break



# Release camera
video.release()
